database/src/pool/postgres.rs

```diff
@@ -743,10 +748,8 @@ impl PostgresConnection {
                         UNION
                         SELECT tag FROM parents
                     )
-                    SELECT job_queue.*
-                    FROM requests
-                    -- Only get requests that have some jobs
-                    RIGHT JOIN job_queue on job_queue.request_tag = requests.tag
+                    -- Only get the jobs of in_progress requests
+                    SELECT * FROM job_queue WHERE job_queue.request_tag IN (SELECT * FROM requests)
```

Comment:
Good find! It is indeed better to select the jobs and filter them by the requests, rather than doing it in the opposite direction. Not sure what I was thinking adding a `RIGHT JOIN` here...

What you wrote is essentially
```sql
SELECT *
FROM job_queue
JOIN requests ON job_queue.request_tag = requests.tag
```
it seems that it generates the same plan. I'd still slightly prefer the `JOIN` version though, as I have more confidence that an index will be used for `JOIN ... ON ...` rather than for `WHERE ... IN ...`.

Reply:
b880a243b3eda94ec2749fed7c6b6ab80e298ad6 :+1: (I've used `INNER JOIN` explicitly)